### ISSUE TITLE
Support for borrowed strings?

### ISSUE THREAD

**Issue by daboross** [dtolnay/serde-ignored#2]
Hi!

I'm not sure what would be required for this, but it would be awesome if borrowed strings could be deserialized with `serde_ignored`. Maybe there's a way to do this that I'm not finding?

My current test case for this:

```rust
extern crate serde;
extern crate serde_json;
extern crate serde_ignored;

use serde::de::Deserialize;

pub fn from_str_with_ignored<'de, T>(input: &'de str) -> Result<T, serde_json::Error>
    where T: Deserialize<'de>
{
    let mut deserializer = serde_json::Deserializer::new(serde_json::de::StrRead::new(input));

    let value = serde_ignored::deserialize(&mut deserializer, |path| {
        println!("unparsed data: {}", path);
    })?;

    deserializer.end()?;

    Ok(value)
}

pub fn from_str_without_ignored<'de, T>(input: &'de str) -> Result<T, serde_json::Error>
    where T: Deserialize<'de>
{
    let mut deserializer = serde_json::Deserializer::new(serde_json::de::StrRead::new(input));

    let value = T::deserialize(&mut deserializer)?;

    deserializer.end()?;

    Ok(value)
}

#[cfg(test)]
mod tests {
    use super::{from_str_with_ignored, from_str_without_ignored};

    #[test]
    fn test_with_serde_ignored() {
        let _: &str = from_str_with_ignored(r#""hi""#).unwrap();
    }
    #[test]
    fn test_without_serde_ignored() {
        let _: &str = from_str_without_ignored(r#""hi""#).unwrap();
    }
}
```

```
running 2 tests
test tests::test_without_serde_ignored ... ok
test tests::test_with_serde_ignored ... FAILED

failures:

---- tests::test_with_serde_ignored stdout ----
	thread 'tests::test_with_serde_ignored' panicked at 'called `Result::unwrap()` on an `Err` value: ErrorImpl { code: Message("invalid type: string \"hi\", expected a borrowed string"), line: 1, column: 4 }', /checkout/src/libcore/result.rs:859
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by daboross** [comment]
Ah, sorry, was working from `crates.io` version.

Seems depending on `serde_ignored = { git = "https://github.com/dtolnay/serde-ignored" }` rather than `serde_ignored = "0.0.3"` fixes this!

Thank you again for making this library, definitely helpful.

**Comment by dtolnay** [comment]
I published a release that includes Alex's fix in 22818c28c6c6e01fe55dedb4cf06353f84f14950.

**Comment by daboross** [comment]
Awesome, thanks!
